src/components/GetData.js

```diff
@@ -0,0 +1,46 @@
+import React, {useEffect, useState} from 'react';
+import axios from "axios";
+
+function GetAllData({search}) {
+    const [result, setResult] = useState({ });
+    const [abilities, setAbilities] = useState([ ]);
+
+
+    const getAbilities = abilities.map((single) => {
+
+        return (
+            <li key={single.slot}>{single.ability.name}</li>
+        )
+    })
+
+    useEffect(() => {
+    async function getData() {
+        try {
+            const response = await axios.get(`https://pokeapi.co/api/v2/pokemon/${search}/`);
+
+            setResult(response.data);
+            setAbilities(response.data.abilities)
+        }
+        catch (error) {
+            console.error(error);
+        }
+    }
+    void getData();
+    }, [search]);
+
```

Comment:
Als je hier
```suggestion
    if (!result) return null;
```
doet, dan kun je de check die hieronder staat weghalen en ben je wat nesting kwijt